src/components/TopBar/Wallet/Assets/TransactionHistory.tsx

```diff
@@ -59,10 +66,14 @@ const TransactionHistory: FC<OHMTransactionHistoryProps> = () => {
   } = GetTransactionHistory();
   const { address, networkId } = useWeb3Context();
   const { data: gOhmTransfers, isFetched: gOhmTransfersFetched } = GetTransferHistory(
-    addresses[networkId].GOHM_ADDRESS,
+    GOHM_ADDRESSES[networkId as keyof typeof GOHM_ADDRESSES],
```

Comment:
Is there a way to make this cleaner? Having to do it every time `STAKING_ADDRESSES` is accessed seems complicated.

Reply:
More of an architectural issue for the constants, rather than this PR.

Reply:
Hey thanks for picking this up @0xJem, i've just realised this code is causing a critical bug that crashes our whole app in production.

`networkId` is sourced from `useWeb3Context`, and is therefore derived from whatever network the users wallet is connected to. A user can connect to *any* network, including networks that aren't supported/recognised in our app. 
Therefore right now it should **never** be passed as a value to get contract addresses.

My design of the new constants are correct. Contracts don't exist on every network, so when we define the addresses for a given contract, we need TypeScript to be able to enforce that we are only accessing an address on a network where that contract *actually* exists. Unfortunately, the types for the old `addresses` don't help to assert this. But when converting it to use the new GOHM_ADDRESSES in this PR, TS could finally enforce a warning that this bug will occur.

You can verify the bug by visiting https://app.olympusdao.finance/#/wallet/history with your wallet connected to an unsupported network, like BSC or something.

I still think it's good to merge this PR since there are no functionality changes. I'll open a separate issue and PR to fix this and avoid "paving over" the bug by typecasting it with `as`.